src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSOutputCommitter.java

```diff
@@ -0,0 +1,149 @@
+    delegate.setupJob(jobContext);
+  }
+
+  @Override
+  public void cleanupJob(JobContext jobContext) throws IOException {
+    delegate.cleanupJob(jobContext);
+  }
+
+  @Override
+  public void commitJob(JobContext jobContext) throws IOException {
+    delegate.commitJob(jobContext);
+  }
+
+  @Override
+  public void abortJob(JobContext jobContext, JobStatus.State state) throws IOException {
+    delegate.abortJob(jobContext, state);
+  }
+
+  @Override
+  public void setupTask(TaskAttemptContext taskAttemptContext) throws IOException {
+    delegate.setupTask(taskAttemptContext);
+  }
+
+  @Override
+  public boolean needsTaskCommit(TaskAttemptContext taskAttemptContext) throws IOException {
+    return delegate.needsTaskCommit(taskAttemptContext);
+  }
+
+  @Override
+  public void commitTask(TaskAttemptContext taskAttemptContext) throws IOException {
+      /*On commit task, there seems to be some inconsistency across different hadoop implementations regarding the path
+       where output file is stored. For some implementations it appears in the path returned by FileOutputCommitter
+       getCommittedTaskPath and for some it does not.Before commit, the files appear to be consistently present in path
+       returned by FileOutputCommitter getTaskAttemptPath. Hence, find the output file from taskAttemptPath and add
+       metadata before commit happens. After commit, file would have been moved out of the taskAttemptPath. */
+    try {
+      updateMetricMetaData(taskAttemptContext);
+    } catch (Exception exception) {
+      LOG.warn("Unable to record metric for task. Metric emitted for the number of affected rows may be incorrect.",
+               exception);
+    }
+
+    delegate.commitTask(taskAttemptContext);
+  }
+
+  private void updateMetricMetaData(TaskAttemptContext taskAttemptContext) throws IOException {
+    if (!(delegate instanceof FileOutputCommitter)) {
+      return;
+    }
+
+    FileOutputCommitter fileOutputCommitter = (FileOutputCommitter) delegate;
+    Configuration configuration = taskAttemptContext.getConfiguration();
+    //Task is not yet committed, so should be available in attempt path
+    Path taskAttemptPath = fileOutputCommitter.getTaskAttemptPath(taskAttemptContext);
+    if (configuration == null || taskAttemptPath == null) {
+      return;
+    }
+
+    //read the count from configuration
+    String keyInConfig = String.format(RECORD_COUNT_FORMAT, taskAttemptContext.getTaskAttemptID());
+    Map<String, String> metaData = new HashMap<>();
+    metaData.put(GCSBatchSink.RECORD_COUNT, String.valueOf(configuration.getLong(keyInConfig, 0L)));
+    StorageClient storageClient = getStorageClient(configuration);
+    //update metadata on the output file present in the directory for this task
+    Blob blob = storageClient.pickABlob(taskAttemptPath.toString());
+    if (blob == null) {
+      LOG.info("Could not find a file in path %s to apply count metadata.", taskAttemptPath.toString());
```

Comment:
`"Could not find a file in path {} to apply count metadata."`

Reply:
PR Updated.